Dockerfile

```diff
@@ -1,10 +1,19 @@
-FROM python:3.5.7-alpine3.10
-RUN apk add git
+ARG PYTHON_VERSION=3.5.7-alpine3.10
```

Comment:
Could we change this to something on the lines of 
```yaml
ARG PYTHON_DOCKER_VERSION=${PYTHON_DOCKER_VERSION:3.5.7-alpine3.10}
```

- a user could use a custom version of Python (if planning on doing an upgrade or some other factors)
- changing the name to make sure the user doesn't have any `DOCKER_VERSION` env variable set-up on his system

Also, should we keep this version of Python or go for a newer one?

Reply:
renaming the variable to `PYTHON_DOCKER_VERSION` makes sense since this is actually a docker tag, but it can be just the python version, it's still valid as a docker tag. So both names work. I'm ok with changing it for now.

The purpose of this arg is to ease testing with various docker images that provide different Python version. If a user wants to use a custom version of python the whole `FROM` needs to be changed. A scenario not intended to be covered by this ARG.

The environment variables set do not transfer over to the `ARG`. That's part of how `ARG` works. You need to pass `docker build --build-arg key=value`.

Changing the python version is out of the scope of this PR.